docs/bigquery_snippets.py

```diff
@@ -0,0 +1,198 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Testable usage examples for Google Cloud BigQuery API wrapper
+
+Each example function takes a ``client`` argument (which must be an instance
+of :class:`gcloud.bigquery.client.Client`) and uses it to perform a task with
+the API.
+
+To facility running the examples as system tests, each example is also passed
+a ``to_delete`` list;  the function adds to the list any objects created which
+need to be deleted during teardown.
+"""
+
+import time
+
+from gcloud.bigquery.client import Client
+
+
+def snippet(func):
+    """Mark ``func`` as a snippet example function."""
+    func._snippet = True
+    return func
+
+
+def _millis():
+    return time.time() * 1000
+
+
+@snippet
+def client_list_datasets(client, to_delete):  # pylint: disable=unused-argument
+    """List datasets for a project."""
+
+    def do_something_with(sub):  # pylint: disable=unused-argument
+        pass
+
+    # [START client_list_datasets]
+    datasets, token = client.list_datasets()   # API request
+    while True:
+        for dataset in datasets:
+            do_something_with(dataset)
+        if token is None:
+            break
+        datasets, token = client.list_datasets(page_token=token)  # API request
+    # [END client_list_datasets]
+
+
+@snippet
+def dataset_create(client, to_delete):
+    """Create a dataset."""
+    DATASET_NAME = 'dataset_create_%d' % (_millis(),)
+
+    # [START dataset_create]
+    dataset = client.dataset(DATASET_NAME)
+    dataset.create()              # API request
+    # [END dataset_create]
+
+    to_delete.append(dataset)
+
+
+@snippet
+def dataset_exists(client, to_delete):
+    """Test existence of a dataset."""
+    DATASET_NAME = 'dataset_exists_%d' % (_millis(),)
+    dataset = client.dataset(DATASET_NAME)
+    to_delete.append(dataset)
+
+    # [START dataset_exists]
+    assert not dataset.exists()   # API request
+    dataset.create()              # API request
+    assert dataset.exists()       # API request
+    # [END dataset_exists]
+
+
+@snippet
+def dataset_reload(client, to_delete):
+    """Reload a dataset's metadata."""
+    DATASET_NAME = 'dataset_reload_%d' % (_millis(),)
+    ORIGINAL_DESCRIPTION = 'Original description'
+    LOCALLY_CHANGED_DESCRIPTION = 'Locally-changed description'
+    dataset = client.dataset(DATASET_NAME)
+    dataset.description = ORIGINAL_DESCRIPTION
+    dataset.create()
+    to_delete.append(dataset)
+
+    # [START dataset_reload]
+    assert dataset.description == ORIGINAL_DESCRIPTION
+    dataset.description = LOCALLY_CHANGED_DESCRIPTION
+    assert dataset.description == LOCALLY_CHANGED_DESCRIPTION
+    dataset.reload()              # API request
+    assert dataset.description == ORIGINAL_DESCRIPTION
```

Comment:
The original didn't show server-side properties replacing local changes (which is the point of `reload`).